csm/core/services/usl.py

```diff
@@ -484,21 +485,39 @@ async def get_system_certificates_by_type(self, material_type: str) -> bytes:
             raise CsmNotFoundError(reason)
         return material
 
-    async def get_network_interfaces(self) -> List[Dict[str, Any]]:
+    async def _get_public_ip(self) -> str:
         """
-        Provides a list of all network interfaces in a system.
+        Reads UDS public IP from global index in an attempt to override UDS default behavior.
+        If it is not found, uses cluster IP as UDS public IP.
 
-        :return: A list containing dictionaries, each containing information about a specific
-            network interface.
+        :return: A string representing UDS public IP address.
         """
+        try:
+            ip = Conf.get(const.CSM_GLOBAL_INDEX, 'UDS.public_ip')
```

Comment:
Here we are getting the value of `UDS.public_ip`. From where the value this value is set ?

Reply:
We do not officially support this configuration option, so it is set by hand. `Conf.get()` will return `None` in this case, and the coercion into a valid IP address in the following line will fail. But as I mentioned on [this comment](https://github.com/Seagate/cortx-manager/pull/330#discussion_r545789634), it is expected to fail.

Reply:
Added new option to csm_setup config, --uds-public-ip, to allow for semi-automatic UDS public IP configuration.